scripts/build/TestImpactAnalysis/Testing/test_tiaf_unit_tests.py

```diff
@@ -0,0 +1,233 @@
+    assert Counter(list1) == Counter(list2)
+
+
+class TestTIAFUnitTests():
+
+    def test_config_files_exist(self, config_path):
+        # given:
+        # config path, and files have been built
+        # when:
+        # we look for the file
+        # then:
+        assert os.path.exists(config_path)
+
+    def test_valid_config(self, caplog, tiaf_args, mock_runtime, mocker, default_runtime_args):
+        # given:
+        # default arguments
+        # when:
+        tiaf = TestImpact(tiaf_args)
+        # then:
+        assert_list_content_equal(
+            tiaf.runtime_args, default_runtime_args.values())
+
+    def test_invalid_config(self, caplog, tiaf_args, mock_runtime, tmp_path_factory, default_runtime_args):
+        # given:
+        invalid_file = tmp_path_factory.mktemp("test") / "test_file.txt"
+        with open(invalid_file, 'w+') as f:
+            f.write("fake json")
+        tiaf_args['config'] = invalid_file
+
+        # when & then:
+        with pytest.raises(SystemError) as exc_info:
+            tiaf = TestImpact(tiaf_args)
+
+    @pytest.mark.parametrize("branch_name", ["development", "not_a_real_branch"])
+    def test_src_branch(self, caplog, tiaf_args, mock_runtime, default_runtime_args, branch_name):
+        # given:
+        tiaf_args['src_branch'] = branch_name
+        # when:
+        tiaf = TestImpact(tiaf_args)
+        # then:
+        assert tiaf.source_branch == branch_name
+
+    @pytest.mark.parametrize("branch_name", ["development", "not_a_real_branch"])
+    def test_dst_branch(self, caplog, tiaf_args, mock_runtime, default_runtime_args, branch_name):
+        # given:
+        tiaf_args['dst_branch'] = branch_name
+        # when:
+        tiaf = TestImpact(tiaf_args)
+
+        # then:
+        assert tiaf.destination_branch == branch_name
+
+    @pytest.mark.parametrize("commit", ["9a15f038807ba8b987c9e689952d9271ef7fd086", "foobar"])
+    def test_commit(self, caplog, tiaf_args, mock_runtime, default_runtime_args, commit):
+        # given:
+        tiaf_args['commit'] = commit
+
+        # when:
+        tiaf = TestImpact(tiaf_args)
+
+        # then:
+        assert tiaf.destination_commit == commit
+
+    def test_no_s3_bucket_name(self, caplog, tiaf_args, mock_runtime, default_runtime_args):
+        # given:
+        # default args
+        test_string = "Attempting to access persistent storage for the commit 'foobar' for suite 'main'"
+
+        # when:
+        tiaf = TestImpact(tiaf_args)
+
+        # then:
+        assert test_string in caplog.messages
+
+    @pytest.mark.parametrize("bucket_name,top_level_dir", [("test_bucket", None), ("test_bucket", "test_dir")])
+    def test_s3_bucket_name(self, caplog, tiaf_args, mock_runtime, default_runtime_args, mocker, bucket_name, top_level_dir, config_data):
+        # given:
+        tiaf_args['s3_bucket'] = bucket_name
+        tiaf_args['s3_top_level_dir'] = top_level_dir
+        mock_storage = mocker.patch(
```

Comment:
You can move any mocked functions at the top of the test function as a decorator. I personally prefer this so the body is more focused on the test, but it's just a suggestion.

Reply:
See https://github.com/o3de/o3de/pull/10757#discussion_r928134889